src/ellipse.rs

```diff
@@ -282,9 +301,149 @@ impl Shape for Ellipse {
     }
 }
 
+/// Calculates circumference C of an ellipse with radii (x, y) as the infinite series
+///
+/// C = pi (x+y) * ∑ binom(1/2, n)^2 * h^n from n = 0 to inf
+/// with h = (x - y)^2 / (x + y)^2
+/// and binom(.,.) the binomial coefficient
+///
+/// as described by Kummer ("Über die Hypergeometrische Reihe", 1837) and rediscovered by
+/// Linderholm and Segal ("An Overlooked Series for the Elliptic Perimeter", 1995).
+///
+/// The series converges very quickly for ellipses with only moderate eccentricity (`h` not close
+/// to 1).
+///
+/// The series is truncated to the sixth power, meaning a lower bound on the true value is
+/// returned. Adding the value of [`kummer_elliptic_perimeter_range`] to the value returned by this
+/// function calculates an upper bound on the true value.
+//
+// Note: Kurbo's minimum supported Rust version doesn't support const floating point arithmetic
+// yet, but this function can be made const on stable Rust.
+#[inline]
+fn kummer_elliptic_perimeter(radii: Vec2) -> f64 {
+    let Vec2 { x, y } = radii;
+    let h = ((x - y) / (x + y)) * ((x - y) / (x + y));
+    let h2 = h * h;
+    let h3 = h2 * h;
+    let h4 = h3 * h;
+    let h5 = h4 * h;
+    let h6 = h5 * h;
+
+    let lower = 1.
+        + h / 4.
+        + h2 / 64.
+        + h3 / 256.
+        + 25. * h4 / 16384.
+        + 49. * h5 / 65536.
+        + 441. * h6 / 1048576.;
+
+    PI * (x + y) * lower
+}
+
+/// This calculates the error range of [`kummer_elliptic_perimeter`]. That function returns a lower
+/// bound on the true value, and though we do not know what the remainder of the infinite series
+/// sums to, we can calculate an upper bound:
+///
+/// ∑ binom(1/2, n)^2 for n = 0 to inf
+///   = 1 + (1 / 2!!)^2 + (1!! / 4!!)^2 + (3!! / 6!!)^2 + (5!! / 8!!)^2 + ..
+///   = 4 / pi
+///   with !! the double factorial
+/// (equation 274 in "Summation of Series", L. B. W. Jolley, 1961).
+///
+/// This means the remainder of the infinite series for C, assuming the series was truncated to the
+/// mth term and h = 1, sums to
+/// 4 / pi - ∑ binom(1/2, n)^2 for n = 0 to m-1
+///
+/// As 0 ≤ h ≤ 1, this is an upper bound.
+//
+// Note: Kurbo's minimum supported Rust version doesn't support const floating point arithmetic
+// yet, but this function can be made const on stable Rust.
+#[inline]
+fn kummer_elliptic_perimeter_range(radii: Vec2) -> f64 {
+    let Vec2 { x, y } = radii;
+    let h = ((x - y) / (x + y)) * ((x - y) / (x + y));
+    let h2 = h * h;
+    let h3 = h2 * h;
+    let h4 = h3 * h;
+    let h5 = h4 * h;
+    let h6 = h5 * h;
+    let h7 = h6 * h;
+
+    const BINOM_SQUARED_REMAINDER: f64 = 0.00101416479131503;
+    // = 4. / PI - (1. + 1. / 4. + 1. / 64. + 1. / 256. + 25. / 16384. + 49. / 65536. + 441. / 1048576.)
+
+    let remainder = BINOM_SQUARED_REMAINDER * h7;
+
+    PI * (x + y) * remainder
```

Comment:
This can be simplified to `(PI * BINOM_SQUARED_REMAINDER) * h.powi(7)` and eliminate the intermediate h's.

Reply:
Ah, true. I originally developed on nightly Rust and marked those functions `const fn` (`f64::powi` isn't available on nightly yet).